pkg/customflag/validate.go

```diff
@@ -0,0 +1,207 @@
+package customflag
+
+import (
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/rancher/distros-test-framework/config"
+	"github.com/rancher/distros-test-framework/shared"
+)
+
+// ValidateTemplateFlags validates version bump template flags that were set on environment variables at .env file.
+func ValidateTemplateFlags() {
+	if err := config.SetEnv(shared.BasePath() + "/config/.env"); err != nil {
+		os.Exit(1)
+	}
+
+	testTag := os.Getenv("TEST_TAG")
+	expectedValue := os.Getenv("EXPECTED_VALUE")
+
+	// validate if expected value was sent because it is required for all tests.
+	if expectedValue == "" {
+		shared.LogLevel("error", "expected value was not sent")
+		os.Exit(1)
+	}
+
+	// validate if flag for install version or commit was sent we should have the expected value after upgrade.
+	instalVersionOrCommit := os.Getenv("INSTALL_VERSION_OR_COMMIT")
+	valuesUpgrade := os.Getenv("VALUE_UPGRADED")
+	if instalVersionOrCommit != "" && valuesUpgrade == "" {
+		shared.LogLevel("error", "using upgrade, please provide the expected value after upgrade")
+		os.Exit(1)
+	}
+
+	expected := strings.Split(expectedValue, ",")
+	expectedUpgrade := strings.Split(valuesUpgrade, ",")
+
+	switch testTag {
+	case "versionbump":
+		validateVersionBumpTest(expected, expectedUpgrade, valuesUpgrade)
+	case "cilium":
+		validateCiliumTest(expected, expectedUpgrade, valuesUpgrade)
+	case "multus":
```

Comment:
Should add one for flannel as well, right? Can we incorporate these func in to one method and add the cmd length on each case statements?
for ex:

```
cmdlen := ""
switch testTag {
	...
	case "cilium":
                    cmdlen = 2
                    validateCNITest(cmdlen, expected, expectedUpgrade, valuesUpgrade)
        case "multus":
                    cmdlen = 4
                    validateCNITest(cmdlen, expected, expectedUpgrade, valuesUpgrade)
         case "flannel":
                     cmdlen = 1
                    validateCNITest(cmdlen, expected, expectedUpgrade, valuesUpgrade)
```
to reduce duplication of code?

Reply:
- for flannel we dont need it, since its a one command x expected value only, it will be validate already on line 22.

- for the cmdLen i really dont see the need of adding as parameter, since it will never change programmatically.